lighthouse.test.js

```diff
@@ -1,8 +1,29 @@
-const lighthouse = require('lighthouse');
-const chromeLauncher = require('chrome-launcher');
+const lighthouse = require('lighthouse')
+const chromeLauncher = require('chrome-launcher')
 const Table = require('cli-table')
+const net = require('net')
 
-let table = new Table()
+const checkConnection = (host, port, timeout = 10000) => {
+  return new Promise((resolve, reject) => {
+    const timer = setTimeout(() => {
+      // eslint-disable-next-line prefer-promise-reject-errors
+      reject('timeout')
+      // eslint-disable-next-line no-use-before-define
+      socket.end()
+    }, timeout)
+    const socket = net.createConnection(port, host, () => {
+      clearTimeout(timer)
+      resolve(true)
+      socket.end()
```

Comment:
Inverse here.

Why not end the socket, and _then_ resolve?

Reply:
I'd also recommend `return resolve` or `return reject` because it tends to be clearer.